demo/simple-browser.lisp

```diff
@@ -50,8 +50,29 @@ Loads and renders a single web page."
      (webkit2:webkit-web-view-load-uri view "http://www.example.com")
      (gtk:gtk-widget-show-all win))))
 
-(defun do-simple-browser-main (&key (private nil))
-  (if private
-      (private-browser-main)
-      (simple-browser-main))
+(defun extended-browser-main ()
+  "A version of `simple-browser-main' that adds a simple WebKitWebExtension."
+  (gtk:within-main-loop
+    (let* ((win (make-instance 'gtk:gtk-window))
+           (manager (make-instance 'webkit:webkit-website-data-manager
+                                   :base-data-directory "testing-data-manager"))
+           (context (make-instance 'webkit:webkit-web-context
+                                   :website-data-manager manager))
+           (view (make-instance 'webkit2:webkit-web-view
+                                :web-context context)))
+      (gobject:g-signal-connect win "destroy"
+                                #'(lambda (widget)
+                                    (declare (ignore widget))
+                                    (gtk:leave-gtk-main)))
+      (webkit:webkit-web-context-set-web-extensions-directory
+                                           context "test-extensions")
```

Comment:
"test-extensions" is supposed to be the path to the library folder, right?
If so, then this line won't set the directory to the right place if the current-directory `uiop:getcwd` is not the "demo" folder.

I suggest deriving this folder from the asd location, e.g. with `(asdf:system-source-directory (asdf:find-system :cl-webkit2))`.